nativelib/src/main/scala/scala/scalanative/native/time.scala

```diff
@@ -0,0 +1,33 @@
+package scala.scalanative
+package native
+
+@extern
+object time {
+
+  // Time manipulation
+
+  def difftime(time_end: time_t, time_beg: time_t): CDouble = extern
+  def time(arg: Ptr[time_t]): time_t                        = extern
+  def clock(): clock_t                                      = extern
+
+  // Format conversions
+
+  def asctime(time_ptr: Ptr[tm]): CString = extern
+  def ctime(time: Ptr[time_t]): CString   = extern
+  def strftime(str: CString, count: CSize, format: CString, time: Ptr[tm]) =
+    extern
+  def gmtime(time: Ptr[time_t]): Ptr[tm]    = extern
+  def localtime(time: Ptr[time_t]): Ptr[tm] = extern
+  def mktime(time: Ptr[tm]): time_t         = extern
+
+  // Types
+
+  type tm      = CStruct9[CInt, CInt, CInt, CInt, CInt, CInt, CInt, CInt, CInt]
+  type time_t  = CLong
+  type clock_t = CLong
```

Comment:
Not quite sure what to do with time_t and clock_t. They are both supposed to be transparent "implementation specific" types - time_t in particular is often a CInt but also often a CLong and as of C11 it isn't even a numeric type at all.

CStruct0 doesn't work, presumably because of #555